plugin.d.ts

```diff
@@ -84,33 +84,38 @@ declare module 'fastify' {
   }
 }
 
-export interface CookieSerializeOptions {
-  domain?: string;
-  encode?(val: string): string;
-  expires?: Date;
-  httpOnly?: boolean;
-  maxAge?: number;
-  path?: string;
-  sameSite?: boolean | 'lax' | 'strict' | 'none';
-  secure?: boolean;
-  signed?: boolean;
-}
-
-interface Signer {
-  sign: (input: string) => string;
-  unsign: (input: string) => {
-    valid: boolean;
-    renew: boolean;
-    value: string | null;
-  };
-}
+declare namespace fastifyCookie {
+  export interface Signer {
+    sign: (input: string) => string;
+    unsign: (input: string) => {
+      valid: boolean;
+      renew: boolean;
+      value: string | null;
+    };
+  }
+  
+  export interface CookieSerializeOptions {
+    domain?: string;
+    encode?(val: string): string;
+    expires?: Date;
+    httpOnly?: boolean;
+    maxAge?: number;
+    path?: string;
+    sameSite?: boolean | 'lax' | 'strict' | 'none';
+    secure?: boolean;
+    signed?: boolean;
+  }
+  
+  export interface FastifyCookieOptions {
+    secret?: string | string[] | Signer;
+    parseOptions?: fastifyCookie.CookieSerializeOptions;
+  }
+  
+  export const fastifyCookie: FastifyPluginCallback<NonNullable<FastifyCookieOptions>>;
 
-export interface FastifyCookieOptions {
-  secret?: string | string[] | Signer;
-  parseOptions?: CookieSerializeOptions;
+  export { fastifyCookie as default };
 }
 
-declare const fastifyCookie: FastifyPluginCallback<NonNullable<FastifyCookieOptions>>;
+declare function fastifyCookie(): FastifyPluginCallback<NonNullable<fastifyCookie.FastifyCookieOptions>>;
```

Comment:
I don't think it is the correct type.

```ts
import defaultFastifyCookie, { fastifyCookie } from "@fastify/cookie";

typeof defaultFastifyCookie === 'function'
typeof defaultFastifyCookie() === 'function'

const isFunc1: Function = defaultFastifyCookie
const isFunc2: Function = defaultFastifyCookie() // it should fail
const isFunc3: Function = fastifyCookie
```

Reply:
yeah, correct point
seems like TS still lacks a way to specify type for function declarations:
https://github.com/microsoft/TypeScript/issues/22063
I'll try to workaround this